### PR TITLE
Jormungandr: Fix stop_areas by lon;lat 

### PR DESCRIPTION
**Pull request by azime**
Jira: http://jira.canaltp.fr/browse/NAVITIAII-2156

### CODE REVIEW

File: source/jormungandr/jormungandr/interfaces/v1/Places.py [hunk 1]
```diff
@@ -359,9 +359,9 @@ def get(self, region=None, lon=None, lat=None):
         # to the region, else, we do a word wide search
 
         if any([region, lon, lat]):
-            instance = i_manager.get_region(region, lon, lat)
-            timezone.set_request_timezone(region)
-            response = i_manager.dispatch(args, "places", instance_name=instance)
+            self.region = i_manager.get_region(region, lon, lat)
+            timezone.set_request_timezone(self.region)
+            response = i_manager.dispatch(args, "places", instance_name=self.region)
```
Comment (TeXitoi) [inline source/jormungandr/jormungandr/interfaces/v1/Places.py:364]: I think you don't need a class member for region here.

File: source/jormungandr/jormungandr/interfaces/v1/ResourceUri.py [hunk 2]
```diff
@@ -66,10 +67,15 @@ def __init__(self, authentication=True, links=True, *args, **kwargs):
             self.method_decorators.append(authentication_required)
 
     def get_filter(self, items, args):
+
         filter_list = [args["filter"]] if args.get("filter") else []
-        if len(items) % 2 != 0:
-            items = items[:-1]
+
         type_ = None
+        if len(items) % 2 != 0:
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/interfaces/v1/ResourceUri.py]: can you add a comment there ?

File: source/jormungandr/tests/authentication_tests.py [hunk 3]
```diff
@@ -27,17 +27,20 @@
...
 from .check_utils import *
 from jormungandr import app
 import json
-from nose.util import *
 import mock
 
+"""
+coords in main_routing_test coverage
+"""
+coords = '{lon};{lat}'.format(lon=0.000628818387368, lat=0.00107797437835)
```
Comment (antoine-de) [inline source/jormungandr/tests/authentication_tests.py]: can't you use https://github.com/CanalTP/navitia/blob/dev/source/jormungandr/tests/check_utils.py#L1124 ?

File: source/jormungandr/tests/authentication_tests.py [hunk 4]
```diff
@@ -419,8 +422,47 @@ def test_sort_coverage(self):
             assert regions[1]["name"] == 'empty routing'
             assert regions[2]["name"] == 'routing api data'
 
+    def test_coverage_by_coords(self):
```
Comment (antoine-de) [inline source/jormungandr/tests/authentication_tests.py:420]: can you add a comment on the different tests?

File: source/jormungandr/tests/authentication_tests.py [hunk 5]
```diff
@@ -419,8 +422,47 @@ def test_sort_coverage(self):
...
+        with user_set(app, FakeUserAuth, 'bobitto'):
+            response = self.query('v1/coverage/{coords}'.format(coords=coords))
+
+            r = get_not_null(response, 'regions')
+
+            region_ids = {region['id']: region for region in r}
```
Comment (antoine-de) [inline source/jormungandr/tests/authentication_tests.py]: minor you can rewrite those 3 lines: `assert {region['id'] for region in r} == {'main_routing_test'}`

File: source/jormungandr/tests/authentication_tests.py (hunk 3 above)
Comment (azime) [reply source/jormungandr/tests/authentication_tests.py]: OK

File: source/jormungandr/tests/authentication_tests.py (hunk 5 above)
Comment (azime) [reply source/jormungandr/tests/authentication_tests.py]: ok

File: source/jormungandr/tests/authentication_tests.py (hunk 4 above)
Comment (azime) [reply source/jormungandr/tests/authentication_tests.py:420]: OK

**Review by antoine-de — APPROVED** [review]

File: source/jormungandr/jormungandr/interfaces/v1/Places.py (hunk 1 above)
Comment (azime) [reply source/jormungandr/jormungandr/interfaces/v1/Places.py:364]: self.region used by ManageStands decorator

**Comment by azime** [comment]
jenkins, test this please

**Pull request merged**